proposals/p3407.md

```diff
@@ -0,0 +1,200 @@
+# Clarify name bindings in namespaces.
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/3407)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Abstract](#abstract)
+-   [Problem](#problem)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+    -   [Allow prefixing a tuple binding pattern with a namespace](#allow-prefixing-a-tuple-binding-pattern-with-a-namespace)
+    -   [Allow binding patterns to declare names in multiple namespaces](#allow-binding-patterns-to-declare-names-in-multiple-namespaces)
+    -   [Allow declaring names in namespaces not owned by the current scope](#allow-declaring-names-in-namespaces-not-owned-by-the-current-scope)
+    -   [Allow declaring namespaces in scopes other than the file scope](#allow-declaring-namespaces-in-scopes-other-than-the-file-scope)
+
+<!-- tocstop -->
+
+## Abstract
+
+-   Require namespace members be declared in the same name scope as the
+    namespace is declared.
+    -   Declaring namespaces outside file scope is disallowed, so this means
+        only at file scope.
+-   Allow binding patterns to directly declare names in namespaces.
+-   Disallow using different namespaces in the same binding pattern.
```

Comment:
```suggestion
-   Disallow introducing bindings in different namespaces in the same pattern.
```
The previous phrasing read to me as if you can't use `N.a: M.T` if `N` and `M` are different namespaces, but you can use `(N.a: N.T, M.b: M.U)` because that's two different binding patterns.